### ISSUE TITLE
[RFC] Button focus outline/glow styles

### ISSUE THREAD

**Issue by brianespinosa** [stardust-ui/react#45]
## Current 
Today when focusing on a button we see the native browser focus styles as seen below in these screenshots:
![image](https://user-images.githubusercontent.com/1161781/43607898-874fab80-9654-11e8-8357-661d4cc84d4a.png)
![image](https://user-images.githubusercontent.com/1161781/43607911-928d6834-9654-11e8-942b-0cdfbb425e1c.png)
![image](https://user-images.githubusercontent.com/1161781/43607924-9d36d842-9654-11e8-8547-280d62788da5.png)

## Proposal
I propose that we remove these default `:focus` styles and normalize the focus style across all browsers. There were similar focus styles in Semantic UI which were slightly different than `:active` and `:hover`.

### Proposed solution
For now I propose that we use the same darken/lighten function ratios that Semantic UI used for changing the default color for these pseudo classes.

As far as I can tell:
default => #E0E1E2
:active => #BABBBC
:hover => #CACBCD
:focus => (same as :hover)
.active => #C0C1C2 (I think the active class/prop is probably already handled today. This is for reference)

For templating purposes I think all styles below "default" should be derived by darken/lighten functions from the default background color.

#### Note
I noticed that the Input component currently does not have a focus style at all. I can open a separate issue for that to discuss, but from an accessibility standpoint we should have those styles there if we have disabled the native styles for :focus.

I understand Stardust is a WIP and very early so apologies if I am stepping on anyone's toes if this is already being worked on. However, if it is not I am happy to open up a PR with some initial work here for discussion.

**Comment by notandrew** [comment]
Yes, they should all be modified at some point. That default focus is really...strong.

**Comment by alinais** [comment]
We are covering this fix with the recent theming changes. Closing this item.
